### PR TITLE
Add array_apply and array_filter, generally improve array handling

### PR DESCRIPTION
**Pull request by m-mohr**
Implements #114

### CODE REVIEW

File: array_contains.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
     "id": "array_contains",
-    "summary": "List contains an element",
-    "description": "Checks whether the array specified for `data` contains the value specified in `value`. Returns `true` if there's a match, otherwise `false`.\n\n**Remarks:**\n\n* To get the index of the element found, use ``array_find()``.\n* All definitions for the process ``eq()`` regarding the comparison of values apply here as well. This includes that data types MUST be checked strictly, for example a string with the content *1* is not equal to the number *1*.\n* An integer *1* is equal to a floating point number *1.0* as `integer` is a sub-type of `number`. Still, this process may return unexpectedly `false` when comparing floating point numbers due to floating point inaccuracy in machine-based computation.\n* Temporal strings are treated as normal strings and MUST NOT be interpreted.",
+    "summary": "Check whether the List contains a value",
```
Comment (soxofaan) [inline array_contains.json]: maybe better:
> Check whether an array contains a given value

File: array_filter.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            "schema": {
+                "type": "array",
+                "items": {
+                    "description": "Any data type is allowed."
+                }
+            },
+            "required": true
+        },
+        "expression": {
+            "description": "An expression that is evaluated against each value in the array. An array element is dropped if the expression returns `false`.",
```
Comment (soxofaan) [inline array_filter.json]: what about null values returned by expression?
maybe explain it positively: 

> Only the array elements where the expression returns `true` are preserved

File: array_filter.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "id": "array_filter",
+    "summary": "Filter an array based on a logical expression",
+    "description": "Filters the array elements based on a logical expression so that afterwards an array is returned that only contains the values conforming to the expression.",
+    "categories": [
+        "arrays",
+        "filter"
+    ],
+    "parameter_order": [
+        "data",
+        "expression"
```
Comment (soxofaan) [inline array_filter.json]: in #128 , the usage of "expression" was replaced by "condition" (for clarity reasons I guess). Maybe also do that here?

File: array_filter.json (hunk 2 above)
Comment (m-mohr) [reply array_filter.json]: Thanks, fixed.

File: array_contains.json (hunk 1 above)
Comment (m-mohr) [reply array_contains.json]: Thanks, fixed.

File: array_filter.json (hunk 3 above)
Comment (m-mohr) [reply array_filter.json]: Good catch, fixed.

**Pull request merged**